restli-server/src/test/java/com/linkedin/restli/internal/server/response/TestBatchFinderResponseBuilder.java

```diff
@@ -341,4 +344,62 @@ private static Foo generateMetaData(Boolean onError)
     Foo foo = new Foo(map);
     return foo;
   }
+
+  @DataProvider(name = "build_uri_test_cases")
+  public Object[][] testBuildCriteriaURIDataProvider() throws RestLiSyntaxException
+  {
+    String[] requestUrls =
+    {
+      "/greetings?bq=searchGreetings&criteria=List((id:1,tone:SINCERE))&message=hello",
+      "/talent/api/talentHiringProjectCandidates?bq=candidates&candidates=List((candidate:urn%3Ali%3Ats_hire_identity%3A88156577))",
+      "/talent/api/talentHiringProjectCandidates?bq=candidates&candidates=List((candidate:urn%3Ali%3Ats_hire_identity%3A88156577))&fields=candidate",
+    };
+
+    String[] batchCriteriaParameterNames =
+    {
+      "criteria",
+      "candidates",
+      "candidates"
+    };
+
+    ProtocolVersion[] protocolVersions =
```

Comment:
Why is this an array instead of a constant? Will we have more values here?

Reply:
Yeah, it is an array because I found we have two protocol versions in the code. So I assume it is needed to test in both protocol versions.

Reply:
The detail is, if there exists some http client and it send a request with a http header saying it is protocol version 100, then our ResourceContextImpl class internally will parse query parameters differently.

Reply:
I understand that we need to support both 1.0 and 2.0, however all your array values are 2.0

Reply:
yes, I met some issues where it only happens in 1.0 so I will include that fix in this PR and then that is complete

Reply:
As we discussed, let us get the fix in for the projection issue with protocol 2.0.0 support and raise a separate PR for 1.0.0 issue.

Reply:
yes, the 1.0.0 part of the issue is long existing one